ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/SpacingAroundColonRuleTest.kt

```diff
@@ -248,15 +248,49 @@ class SpacingAroundColonRuleTest {
             class X :
                 Y,
                 Z
-            class A // comment
-                : B
-            class A /*
+            class A : // comment
+                B
+            class A : /*
 
             */
-                : B
+                B
             val xmlFormatter:
                 String = ""
             """.trimIndent()
         )
     }
+
+    // https://github.com/pinterest/ktlint/issues/1057
+    @Test
+    fun testLintWithEolComment() {
+        assertThat(
+            SpacingAroundColonRule().lint(
+                """
+                val x // comment
+                    : Int = 1
+                """.trimIndent()
+            )
+        ).isEqualTo(
+            listOf(
+                LintError(1, 17, "colon-spacing", "Unexpected newline before \":\""),
+            )
+        )
+    }
+
+    @Test
+    fun testFormatWithEolComment() {
+        assertThat(
+            SpacingAroundColonRule().format(
+                """
+                val x // comment
+                    : Int = 1
+                """.trimIndent()
+            )
+        ).isEqualTo(
+            """
+                val x: // comment
+                    Int = 1
```

Comment:
Not sure it is the best way to format :thinking: 
Property type should be, imho, on the same line with property name. Probably formatted code could look like this:
```kotlin
val x: Int = // comment
    1
```

or like this:
```kotlin
// comment
val x: Int = 1
```

What do you think?

Reply:
> ```kotlin
> val x: Int = // comment
>     1
> ```
I think this is better.
Fixed in 9460b58.